### PR TITLE
minor changes and docs

### PR DESCRIPTION
**Pull request by Avuxo**
README is a skeleton, I've got some notes I took on paper I'm gonna try to figure out how to put into words

### CODE REVIEW

File: src/mkbktr.cpp [hunk 1]
```diff
@@ -258,13 +258,13 @@ int main(int argc, char **argv) {
...
   NcaFsHeader *fs_headers[4] = {
       reinterpret_cast<NcaFsHeader *>(
-          &patch_header_plaintext[0x400 + 0x200 * 0]),
+          &patch_header_plaintext[0x400 + (0x200 * 0)]),
       reinterpret_cast<NcaFsHeader *>(
-          &patch_header_plaintext[0x400 + 0x200 * 1]),
+          &patch_header_plaintext[0x400 + (0x200 * 1)]),
       reinterpret_cast<NcaFsHeader *>(
-          &patch_header_plaintext[0x400 + 0x200 * 2]),
+          &patch_header_plaintext[0x400 + (0x200 * 2)]),
```
Comment (Avuxo) [inline src/mkbktr.cpp:265]: at first i thought this was a buffer over-run before i remembered PEMDAS, so I just tried to make it more obvious for idiots of my caliber

File: README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+TODO: add guide for small utils
+
+## Building
+
+Building is the standard CMake process:
+```
+mkdir build
+cd build
+ccmake ../
```
Comment (rschlaikjer) [inline README.md:18]: Typo? Or does `ccmake` work?

File: src/mkbktr.cpp [hunk 3]
```diff
@@ -49,19 +49,19 @@ static void nca_update_ctr(uint8_t *ctr, uint64_t offset) {
...
+  if (argc != 5) {
+    fprintf(stderr, "Usage: %s [original nca] [patched nca] [output nca] [/path/to/prod.keys]\n", argv[0]);
     return -1;
   }
 
   // Name args
   const char *original_nca_filename = argv[1];
   const char *new_nca_filename = argv[2];
   const char *bktr_nca_filename = argv[3];
+  const char *path_to_prod_keys = argv[4]; // TODO: this currently needs to be an absolute path.
```
Comment (rschlaikjer) [inline src/mkbktr.cpp]: Does it? Relative paths should work?

Comment (Avuxo) [reply src/mkbktr.cpp]: just a bad comment, i forgot to remove it

File: src/tools/nca_parse.cpp [hunk 4]
```diff
@@ -37,8 +37,8 @@ int main(int argc, char **argv) {
   }
 
   // Load keys
-  auto keys =
-      mk::Keys::from_file("/home/ross/.local/share/yuzu/keys/prod.keys");
+  const char *path_to_prod_keys = argv[2]; // TODO: don't wanna work with absolute paths.
```
Comment (rschlaikjer) [inline src/tools/nca_parse.cpp]: Not sure what you mean by the comment here

File: src/util/mem.cpp [hunk 5]
```diff
@@ -29,7 +29,7 @@ std::unique_ptr<MappedData> MappedData::from_file(const char *filename) {
   }
 
   void *mmapped_data =
-      mmap(nullptr, file_size, PROT_READ, MAP_PRIVATE, file_fd, 0);
+    mmap(nullptr, file_size, PROT_READ, MAP_SHARED, file_fd, 0);
```
Comment (rschlaikjer) [inline src/util/mem.cpp]: Is there any benefit to this? We aren't using this mapped region to share data among threads?

File: README.md (hunk 2 above)
Comment (Avuxo) [reply README.md:18]: ccmake is the curses version of cmake that is friendly and prompts you for errors. I'm willing to swap it back to cmake, but I'm not great at remembering the cache commands so I just tend towards the interactive version.

File: src/util/mem.cpp (hunk 5 above)
Comment (Avuxo) [reply src/util/mem.cpp]: the man page doesn't completely agree with [this SO post](https://stackoverflow.com/questions/7222164/mmap-an-entire-large-file), but I did some experimenting and it appears to be right (at least on WSL)

Comment (rschlaikjer) [reply src/util/mem.cpp]: Hm, fair enough. Can't hurt.

**Pull request merged**